api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +345,147 @@ type AzureWebhookReceiverConfig struct {
+
+	// MatchConditions is a list of conditions that must be met for the action to be performed.
+	//
+	// +optional
+	MatchConditions []ConditionSelector `json:"matchConditions,omitempty" protobuf:"bytes,2,rep,name=matchConditions"`
+
+	// Parameters contains additional parameters for the action.
+	//
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
+
+	// Targets is a list of targets on which the action should be performed.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Targets []GenericWebhookTarget `json:"targets,omitempty" protobuf:"bytes,4,rep,name=targets"`
+}
+
+// GenericWebhookActionName represents the name of an action to be performed on a resource.
+type GenericWebhookActionName string
+
+const (
+	// GenericWebhookActionNameRefresh indicates a request to refresh the resource.
+	GenericWebhookActionNameRefresh GenericWebhookActionName = "Refresh"
+)
+
+// GenericWebhookTarget describes a target resource for a generic webhook event.
+type GenericWebhookTarget struct {
+	// Kind is the kind of the target resource (e.g., "Warehouse", "Stage").
+	//
+	// +kubebuilder:validation:Required
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// LabelSelector is a label selector to identify the target resources.
+	//
+	// +optional
+	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,2,opt,name=labelSelector"`
+
+	// IndexSelector is an index selector to identify the target resources.
+	//
+	// +optional
+	IndexSelector IndexSelector `json:"indexSelector,omitempty" protobuf:"bytes,3,opt,name=indexSelector"`
+}
+
+// IndexSelector encapsulates a selector used to derive index keys
+// based on expressions.
+type IndexSelector struct {
+	// MatchExpressions is a list of index selector requirements.
+	//
+	// +kubebuilder:validation:MinItems=1
+	MatchExpressions []IndexSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,1,rep,name=matchExpressions"`
+}
+
+// IndexSelectorRequirement encapsulates a requirement used to select indexes
+// based on specific criteria.
+type IndexSelectorRequirement struct {
+	// Key is the key of the index.
+	//
+	// kubebuilder:validation:Required
+	Key string `json:"key" protobuf:"bytes,1,opt,name=key"`
+
+	// Operator indicates the operation that should be used to evaluate
+	// whether the selection requirement is satisfied.
+	//
+	// kubebuilder:validation:Required
+	Operator Operator `json:"operator" protobuf:"bytes,2,opt,name=operator"`
+
+	// Values is a list of values or a single value returned from an expression.
+	//
+	// kubebuilder:validation:Required
+	Values apiextensionsv1.JSON `json:"values" protobuf:"bytes,3,opt,name=values"`
+}
+
+// ConditionSelector encapsulates a condition used to match resources based
+// on specific criteria.
```

Comment:
I could be wrong, but I think this description differs from what this is actually used for. Doesn't this have nothing to do with identifying resources and has only to do with determining whether an action will be taken at all?